### PR TITLE
Make registration path mandatory, remove kubeconfig parameter

### PR DESCRIPTION
**Pull request by jarrpa**
Related #8 

Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

**Comment by jarrpa** [comment]
I didn't see much documented in the README so I left it alone for now. If it should be expanded somehow, that could be a separate PR.

File: cmd/node-driver-registrar/node_register.go [hunk 1]
```diff
@@ -65,87 +64,42 @@ func nodeRegister(
...
 
-		// This program is intended to run as a side-car container inside a
-		// Kubernetes DaemonSet. Kubernetes DaemonSet only have one RestartPolicy,
-		// always, meaning as soon as this container terminates, it will be started
-		// again. Therefore, this program will loop indefientley and periodically
-		// update the node annotation.
-		// The CSI driver name and node ID are assumed to be immutable, and are not
-		// refetched on subsequent loop iterations.
-		for {
-			err := getVerifyAndAddNodeId(
```
Comment (msau42) [inline cmd/node-driver-registrar/node_register.go]: Can you remove these dead functions too?

**Comment by jarrpa** [comment]
@msau42 Removed dead functions, fixed test errors.

**Comment by msau42** [comment]
/lgtm
/approve

As a followup, could you cleanup vendor? I think now that these k8s dependencies are removed, we should be able to remove a lot of k8s vendored pkgs.

**Pull request merged**